### DISCUSSION TITLE
d2i PublicKey(); usage error

### DISCUSSION THREAD

**Discussion by northpole55** [openssl/openssl#24154] [Q&A]
Demo rsa_pss_hash.c: d2i_PublicKey(EVP_PKEY_RSA, NULL, &ppub_key, sizeof(rsa_pub_key)); Parsing the public key outside of demo will return an empty value;err_code = ERR_get_error()        err_code  = 0x068000a8;

**Comment by kulkarniamit** [comment]
`ppub_key` used in the demo points to the public key `rsa_pub_key`
https://github.com/openssl/openssl/blob/6a4a714045415be6720f4165c4d70a0ff229a26a/demos/signature/rsa_pss_hash.c#L123-L124
`rsa_pub_key` is stored in `rsa_pss.h`
https://github.com/openssl/openssl/blob/6a4a714045415be6720f4165c4d70a0ff229a26a/demos/signature/rsa_pss.h#L211-L214

So, decoding the public key outside of demo will require pointing `ppub_key` to a valid RSA pub key value in DER format.
